rct229/rulesets/ashrae9012019/section19/section19rule25.py

```diff
@@ -0,0 +1,125 @@
+            rmrs_used=UserBaselineProposedVals(True, True, True),
+            each_rule=Section19Rule25.HVACRule(),
+            index_rmr="baseline",
+            id="19-25",
+            description="Schedules for HVAC fans that provide outdoor air for ventilation shall run continuously whenever spaces are occupied in the baseline design.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1-4 Schedules for the proposed building excluding exception #1 and Section G3.1.2.4.",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmi_u = context.user
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+
+        HVAC_systems_virtual_list_p = list(
+            set(
+                get_proposed_hvac_modeled_with_virtual_cooling(rmi_u, rmi_p)
+                + get_proposed_hvac_modeled_with_virtual_heating(rmi_u, rmi_p)
+            )
+        )
+
+        zones_virtual_heating_cooling_list_p = list(
+            set(
+                [
+                    zone_id_p
+                    for hvac_id_p in HVAC_systems_virtual_list_p
+                    for zone_id_p in get_hvac_zone_list_w_area_dict(rmi_p)[hvac_id_p][
+                        "zone_list"
+                    ]
+                ]
+            )
+        )
+
+        inapplicable_hvac_with_virtual_heating_cooling_list_b = list(
+            set(
+                list(
+                    itertools.chain(
+                        *[
+                            get_list_hvac_systems_associated_with_zone(rmi_b, zone_id_p)
+                            for zone_id_p in zones_virtual_heating_cooling_list_p
+                        ]
+                    )
+                )
+            )
+        )
+
+        return {
+            "inapplicable_hvac_with_virtual_heating_cooling_list_b ": inapplicable_hvac_with_virtual_heating_cooling_list_b
+        }
+
+    class HVACRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section19Rule25.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(True, True, True),
+            )
+
+        def is_applicable(self, context, data=None):
+            hvac_b = context.baseline
+            hvac_id_b = hvac_b["id"]
+            inapplicable_hvac_with_virtual_heating_cooling_list_b = data[
+                "inapplicable_hvac_with_virtual_heating_cooling_list_b"
+            ]
+
+            return (
+                hvac_id_b not in inapplicable_hvac_with_virtual_heating_cooling_list_b
+            )
+
+        def get_calc_vals(self, context, data=None):
+            hvac_b = context.baseline
+
+            operation_during_occupied_b = getattr_(
+                hvac_b, "HVAC", "fan_system", "operation_during_occupied"
+            )
```

Comment:
I see this in many other rules. You can also put them in the `required_fields`.

Reply:
I changed to use the ```required_fields``` instead of ```getattr_```. I'll also the same change in other rules.